### ISSUE TITLE
API routes return empty objects

### ISSUE THREAD

**Issue by nnelgxorz** [BuilderIO/qwik#903]
### Qwik Version

0.0.39

### Operating System (or Browser)

Debian Linux

### Node Version (if applicable)

v16.3.0

### Which component is affected?

Qwik City

### Expected Behaviour

I should be able to call an api endpoint ie `routes/api/get-data.ts` and retrieve a json response

### Actual Behaviour

Response is always an empty object `{}` regardless of what the endpoint is returning.

### Additional Information

Qwik City 0.0.29

**Comment by youngboy** [comment]
+1, same here. with qwik-city@0.0.29. 

if do something async in json endpoint, then will be empty

in [`packages/qwik-city/runtime/src/app/routes/`](https://github.com/BuilderIO/qwik/blob/main/packages/qwik-city/runtime/src/app/routes/api/data.json.ts#L4)  change `onGet` to following then reproduce the issue.

```ts
export const onGet: EndpointHandler = async ({ request }) => {
  await Promise.resolve()
  return {
    timestamp: Date.now(),
    method: request.method,
    url: request.url,
    os: os.platform(),
    arch: os.arch(),
    node: process.versions.node,
  }
};
```

**Comment by nnelgxorz** [comment]
@youngboy I guess that means it's fixed for the. Ext release. 😆

**Comment by nnelgxorz** [comment]
Fixed in 0.0.30
